### DISCUSSION TITLE
OSC Customization - Progress Bar

### DISCUSSION THREAD

**Discussion by advice2020** [mpv-player/mpv#14492] [Q&A]
Linux Debian based distro

`GENERAL INFORMATION`
For now I am just looking to create a very simple OSC that just consists of modifying the default OSC.  I was able to make a lot more progress than I expected but ran into some items that I do not know how to modify.  Most of the issues have to do with dimensions of items, I imagine they are based on percentages rather than a specific values.
I am looking for users that have some knowledge about modifying OSC that hopefully can answer some of my questions.
To start I copied the “osc.lua” file from MPV source code to start with as my base.
I do not have a osc.conf created.
I have the following at the bottom of my “mpv.conf” not sure if these conflict with anything
```
fs=yes
script-opts=osc-scalewindowed=1.50,osc-scalefullscreen=3.00
script-opts=osc-timetotal=yes
osc=no
```
I am aware that there are a couple of 3rd party OSCs available but none that I am looking for.
I just wanted a simple timecode & seekbar layout using the EXACT same items provided in the default osc.lua file "bottombar"
TIMECODE |=========SEEKBAR=========| TIMECODE

Does anyone know how to.....
-center the progress bar horizontally?
-center the progress bar vertically? (progress bar is at the bottom, just wanted to move it up a little)

-move the timecode text
-change the width of the progress bar?
-change the height of the progress bar?

Thank you to anyone who reads this and for any help provided

**Comment by advice2020** [comment]
I am still interested in this if anyone has any ideas.

**Comment by advice2020** [comment]
I might just be asking questions that are too complex / not possible
I thought because MPV has so many customization options and the fact that I imagine it has a huge user base that my questions would get tons of responses.
Does anyone know if the Discussions section here is considered to be the MPV forum or is there an official forum?
Not sure where else to ask these OSC customization questions, only looking to slightly modifying existing OSC, nothing fancy like other designers have achieved with modifying the OSC.

**Comment by Samillion** [comment]
Copying `osc.lua` would definitely do the job, but it can be overwhelming. Especially since you want to make something simple.

If I may suggest, you should check this project https://github.com/maoiscat/mpv-osc-framework

It literally helps you generate a barebones of `osc.lua`. No element management, rendering or anything. You can then slowly add the things you need from `osc.lua` to achieve the final result you desire.

I realize you wouldn't want suggestions of already made OSCs, but, if your goal is a very simple seek OSC, there is one that actually quite closely meets your request:
https://codeberg.org/NRK/mpv-toolbox/src/branch/master/mfpbar

**Comment by advice2020** [comment]
@Samillion 
Thank you in general for responding to all my OSC based posts, I really appreciate it I have found out that there are not many who know anything about this or if they do, that are willing to help with it.
I have actually been able to make some progress after making this original post.
I had seen this "mpv-osc-framework" in the past when I started but to me at the time seemed just as complex, I figured I would just edit the osc.lua file because it already had everything I wanted, little did I know how complicated it would be :)
Even though I have made some progress the methods I used were sketchy so I still do not know what I am doing :)
I kind of only have one thing I could not figure out, which was the reason for "geo" post you commented on, I will mention the post if interested.
Thank you again
